lib/Frontend/nvqpp/ConvertExpr.cpp

```diff
@@ -1948,6 +1984,45 @@ bool QuakeBridgeVisitor::VisitCXXConstructExpr(clang::CXXConstructExpr *x) {
         assert((isa<quake::RefType, quake::VeqType>(tosTy)));
         return true;
       }
+
+      if (ctorName == "vector") {
+        // This is a std::vector constructor, first we'll check if it
+        // is constructed from a constant initializer list, in that case
+        // we'll have a AllocaOp at the top of the stack that allocates a
+        // ptr<array<TxC>>, where C is constant / known
+        if (auto alloca = dyn_cast<cc::AllocaOp>(peekValue().getDefiningOp())) {
```

Comment:
I suspect you need to check the number and type of arguments to this variant of the `std::vector` ctor overloads. In general, we need to know by looking at the AST what *should* be on the stack. Then look at (either) stack to get it and verify that it is what is expected.

Here the difference between an `cc::AllocaOp` and say a `cc::ComputePtrOp` is not so interesting. But if we know this is the init-list variant, then we better have some buffer expression with a `!cc.ptr` type.

(There isn't any code that does this in the original `std::function` case above either, which it a bug that should be fixed. <sigh>)

Reply:
What about updating to something like this 
```cpp
// InitList Vector Ctor has the InitList and Allocator arguments
// so check there are 2, and that the first is an ElaboratedType, which 
// InitList is...
if (x->getNumArgs() == 2 &&
    dyn_cast<clang::ElaboratedType>(x->getArg(0)->getType())) {
  if (auto alloca =
    dyn_cast<cc::AllocaOp>(peekValue().getDefiningOp())) {
       auto asPtrType =
           dyn_cast<cc::PointerType>(alloca.getAddress().getType());
       if (auto arrayTy =
               dyn_cast<cc::ArrayType>(asPtrType.getElementType()))
         if (alloca.getNumOperands() > 0)
             return pushValue(builder.create<cc::StdvecInitOp>(
                 loc, cc::StdvecType::get(arrayTy.getElementType()), alloca,
                 alloca.getSeqSize()));
  }   
}
```

Is this what you mean

Reply:
See latest update.